restapi/README-docsapi.md

```diff
@@ -0,0 +1,168 @@
+# Stargate Documents API
+
+The Documents API is built into the REST API module, an OSGi bundle that exposes various CRUD services over HTTP.
+Specifically, the Documents API allows for storage of documents in the JSON format using C* tables (called "collections" in this context).
+
+Here is a brief overview of the service (for more details, refer to the [Stargate
+online docs]).
+
+* There are endpoints for writing, updating, and deleting documents in [DocumentResource]
+* For reading documents and searching through documents and collections, there are endpoints in [ReactiveDocumentResource]
+* CRUD operations for collections exist in [CollectionsResource], and operations for modifying keyspace information (called "namespaces" in this context) are in [NamespacesResource]
+* Operations for attaching a JSON Schema (see [Json schema info] for more details) are in [JsonSchemaResource]
+
+## Entry points
+
+[RestApiActivator] manages the starting and stopping of the OSGi bundle, which runs both the Documents API and the more general REST API.
+
+[Server] is the HTTP server that exposes the Documents API services as REST resources. It is implemented with
+[Dropwizard](https://www.dropwizard.io/en/latest/).
+
+### Authentication
+
+Every HTTP request performs token-based authentication. This is done by calling [Db]#getDocDataStoreForToken explicitly.
+This method will either return an instance of [DocumentDB] which can be used for querying Cassandra and contains the authenticated subject,
+or will throw an UnauthorizedException causing the appropriate 4XX response to be returned.
+
+The DocumentDB gets stored in a simple cache (using [Caffeine](https://github.com/ben-manes/caffeine)) so the subsequent requests within a short
+time-frame using the same token and headers need not suffer performance penalties.
+
+### Resources
+
+#### Document CUD operations
+
+Note: All endpoints that take a body require a `Content-Type` header of `application/json`.
+
+[DocumentResource] contains resources for creating, updating, and deleting documents.
+If the document to be created is in a collection that doesn't yet exist, it will be created
+before the document is inserted. Here is an overview of each endpoint:
+* postDoc
+  - Create a new document in a collection. A UUID will be assigned as the key for the new document.
+* writeManyDocs
+  - Create many new documents using a JSON array as the body. By default a UUID will be assigned as the key for each document, unless an `id-path` query param is specified.
+* putDoc
+  - Upsert a document. The key for the document will be the value of `:documentId`. This is idempotent and will overwrite any document that exists with the given key.
```

Comment:
Curious, why is it ":documentId" instead of "documentId"?

Reply:
Since I had the external routes written out before (e.g. /api/v2/namespace/:namespace/collections/:collection/:documentId) I was referring to the way I had it written

But then I deleted the external routes and accidentally left the colon in...but now I'm going to re-add them so I'll re-edit this to make sense